### PR TITLE
Side-step pending deletes check

### LINKED ISSUE #30416: [CI] SplitIndexIT fails with Lucene-7.4 snapshot
This test starts failing more often after upgrading to Lucene-7.4 snapshot. Most of the failures are on Windows. It may relate to this Lucene failure https://elasticsearch-ci.elastic.co/job/apache+lucene-solr+master/27765/consoleText.

Error: `still has pending deleted files; cannot initialize IndexWriter`

```
  1> Caused by: java.lang.IllegalArgumentException: Directory store(MMapDirectory@C:\Users\jenkins\workspace\elastic+elasticsearch+master+multijob-windows-compatibility\server\build\testrun\integTest\J1\temp\org.elasticsearch.action.admin.indices.create.SplitIndexIT_734B52BB31887C24-001\tempDir-002\data\nodes\1\indices\Ru5mKdhxQpaT3gWmSTvH2w\16\index lockFactory=org.apache.lucene.store.NativeFSLockFactory@66502081) still has pending deleted files; cannot initialize IndexWriter
  1> 	at org.apache.lucene.index.IndexWriter.<init>(IndexWriter.java:698) ~[lucene-core-7.4.0-snapshot-1ed95c097b.jar:7.4.0-snapshot-1ed95c097b 1ed95c097b82ee5f175e93f3fe62572abe064da6 - jimczi - 2018-05-03 16:16:13]
  1> 	at org.elasticsearch.index.store.Store.newIndexWriter(Store.java:1635) ~[main/:?]
  1> 	at org.elasticsearch.index.store.Store.bootstrapNewHistory(Store.java:1487) ~[main/:?]
  1> 	at org.elasticsearch.index.shard.StoreRecovery.internalRecoverFromStore(StoreRecovery.java:393) ~[main/:?]
  1> 	at org.elasticsearch.index.shard.StoreRecovery.lambda$recoverFromLocalShards$3(StoreRecovery.java:133) ~[main/:?]
  1> 	at org.elasticsearch.index.shard.StoreRecovery.executeRecovery(StoreRecovery.java:301) ~[main/:?]
  1> 	at org.elasticsearch.index.shard.StoreRecovery.recoverFromLocalShards(StoreRecovery.java:123) ~[main/:?]
  1> 	at org.elasticsearch.index.shard.IndexShard.recoverFromLocalShards(IndexShard.java:1520) ~[main/:?]
  1> 	at org.elasticsearch.index.shard.IndexShard.lambda$startRecovery$8(IndexShard.java:2042) ~[main/:?]
  1> 	... 4 more
```

Some CI instances:

1. https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+multijob-windows-compatibility/1608/consoleText

2. https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+multijob-windows-compatibility/1609/console

Log: 
[shrink.txt](https://github.com/elastic/elasticsearch/files/1978298/shrink.txt)

### LINKED ISSUE #30503: [CI] IntegTestZipClientYamlTestSuiteIT indices.split fails on windows tests
Two rest tests keep failing due to timeouts on multijob-windows-compatibility.

links:

- https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+multijob-windows-compatibility/1620/console
- https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.x+multijob-windows-compatibility/1047/console

reproduce:

```
./gradlew :distribution:archives:integ-test-zip:integTestRunner \
  -Dtests.seed=F724E83BFA960B1D \
  -Dtests.class=org.elasticsearch.test.rest.IntegTestZipClientYamlTestSuiteIT \
  -Dtests.method="test {p0=indices.split/20_source_mapping/Split index ignores target template mapping}" \
  -Dtests.security.manager=true \
  -Dtests.locale=it-CH \
  -Dtests.timezone=Etc/GMT-7

./gradlew :distribution:archives:integ-test-zip:integTestRunner \
  -Dtests.seed=F724E83BFA960B1D \
  -Dtests.class=org.elasticsearch.test.rest.IntegTestZipClientYamlTestSuiteIT \
  -Dtests.method="test {p0=indices.split/10_basic/Split from 1 to N}" \
  -Dtests.security.manager=true \
  -Dtests.locale=it-CH \
  -Dtests.timezone=Etc/GMT-7
```

Only `test {p0=indices.split/20_source_mapping/Split index ignores target template mapping}` fails on `6.x`, both fail on `master`

stacktrace:

```
  1> [2018-05-10T06:02:54,341][INFO ][o.e.t.r.IntegTestZipClientYamlTestSuiteIT] [test {p0=indices.split/20_source_mapping/Split index ignores target template mapping}]: before test

  1> [2018-05-10T06:03:24,426][INFO ][o.e.t.r.IntegTestZipClientYamlTestSuiteIT] Stash dump on test failure [{
  1>   "stash" : {
  1>     "body" : {
  1>       "cluster_name" : "distribution_archives_integ-test-zip_integTestCluster",
  1>       "status" : "green",
  1>       "timed_out" : false,
  1>       "number_of_nodes" : 1,
  1>       "number_of_data_nodes" : 1,
  1>       "active_primary_shards" : 1,
  1>       "active_shards" : 1,
  1>       "relocating_shards" : 0,
  1>       "initializing_shards" : 0,
  1>       "unassigned_shards" : 0,
  1>       "delayed_unassigned_shards" : 0,
  1>       "number_of_pending_tasks" : 0,
  1>       "number_of_in_flight_fetch" : 0,
  1>       "task_max_waiting_in_queue_millis" : 0,
  1>       "active_shards_percent_as_number" : 100.0
  1>     }
  1>   }
  1> }]

  1> [2018-05-10T06:03:24,482][INFO ][o.e.t.r.IntegTestZipClientYamlTestSuiteIT] [test {p0=indices.split/20_source_mapping/Split index ignores target template mapping}]: after test

  2> REPRODUCE WITH: gradlew :distribution:archives:integ-test-zip:integTestRunner -Dtests.seed=F724E83BFA960B1D -Dtests.class=org.elasticsearch.test.rest.IntegTestZipClientYamlTestSuiteIT -Dtests.method="test {p0=indices.split/20_source_mapping/Split index ignores target template mapping}" -Dtests.security.manager=true -Dtests.locale=it-CH -Dtests.timezone=Etc/GMT-7

ERROR   30.2s | IntegTestZipClientYamlTestSuiteIT.test {p0=indices.split/20_source_mapping/Split index ignores target template mapping} <<< FAILURES!
   > Throwable #1: java.lang.RuntimeException: Failure at [indices.split/20_source_mapping:57]: listener timeout after waiting for [30000] ms
   > 	at __randomizedtesting.SeedInfo.seed([F724E83BFA960B1D:7F70D7E1546A66E5]:0)
   > 	at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:358)
   > 	at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.test(ESClientYamlSuiteTestCase.java:333)
   > 	at java.lang.Thread.run(Thread.java:748)
   > Caused by: java.io.IOException: listener timeout after waiting for [30000] ms
   > 	at org.elasticsearch.client.RestClient$SyncResponseListener.get(RestClient.java:794)
   > 	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:174)
   > 	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:268)
   > 	at org.elasticsearch.test.rest.yaml.ClientYamlTestClient.callApi(ClientYamlTestClient.java:173)
   > 	at org.elasticsearch.test.rest.yaml.ClientYamlTestExecutionContext.callApiInternal(ClientYamlTestExecutionContext.java:158)
   > 	at org.elasticsearch.test.rest.yaml.ClientYamlTestExecutionContext.callApi(ClientYamlTestExecutionContext.java:90)
   > 	at org.elasticsearch.test.rest.yaml.section.DoSection.execute(DoSection.java:223)
   > 	at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:349)
   > 	... 37 more
```

### PR DESCRIPTION
**Pull request by s1monw**
When we split/shrink an index we open several IndexWriter instances
causeing file-deletes to be pending on windows. This subsequently fails
when we open an IW to bootstrap the index history due to pending deletes.
This change sidesteps the check since we know our history goes forward
in terms of files and segments.

Closes #30416, closes #30503

### CODE REVIEW

**Comment by s1monw** [comment]
this is an alternative solution to #30567

File: server/src/main/java/org/elasticsearch/index/store/Store.java [hunk 1]
```diff
@@ -759,6 +759,17 @@ private void innerClose() throws IOException {
         public String toString() {
             return "store(" + in.toString() + ")";
         }
+
+        @Override
+        public boolean checkPendingDeletions() throws IOException {
+            if (super.checkPendingDeletions() == false) {
+                deletesLogger.debug("directory has still pending deletes");
```
Comment (jpountz) [inline server/src/main/java/org/elasticsearch/index/store/Store.java]: should we warn rather than debug?

**Review by jpountz — APPROVED** [review]

File: server/src/main/java/org/elasticsearch/index/store/Store.java (hunk 1 above)
Comment (s1monw) [reply server/src/main/java/org/elasticsearch/index/store/Store.java]: fair enough.

**Comment by jasontedor** [comment]
I think this also closes #30503 and there are two additional skips that need to be removed:
 - https://github.com/tbrooks8/elasticsearch/blob/470508ab17b4288e633eae06df4e1d8bba86fb3f/rest-api-spec/src/main/resources/rest-api-spec/test/indices.split/20_source_mapping.yml#L6
 - https://github.com/tbrooks8/elasticsearch/blob/470508ab17b4288e633eae06df4e1d8bba86fb3f/rest-api-spec/src/main/resources/rest-api-spec/test/indices.split/10_basic.yml#L112

**Review by jasontedor — APPROVED** [review]
LGTM.

**Comment by s1monw** [comment]
@elasticmachine what are you doing all day?

**Pull request merged**